services/directions.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+
+/**
+ * Directions API service.
+ */
+var Directions = {};
+
+/**
+ * Get directions.
+ *
+ * See the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#directions).
+ *
+ * @param {Object} config
+ * @param {'driving-traffic'|'driving'|'walking'|'cycling'} [config.profile=driving]
+ * @param {Array.<Object>} config.way_points - An ordered array of objects with `latitude`
+ * and `longitude` properties that represent points to visit and optional `approach`, `bearing`, `radius` & `waypoint_name`,
+ * see the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#retrieve-directions) for details.
+ * There can be between 2 and 25 coordinates.
+ * @param {string} [config.alternatives=false]
+ * @param {Array.<string>} [config.annotations]
+ * @param {boolean} [config.banner_instructions=false]
+ * @param {boolean} [config.continue_straight]
+ * @param {string} [config.exclude]
```

Comment:
For each of these with a finite set of options, let's specify the strings that are available (if the list is reasonably short).